core/src/gl/shaderProgram.cpp

```diff
@@ -84,15 +120,20 @@ GLint ShaderProgram::getAttribLocation(const std::string& _attribName) {
 
 }
 
-GLint ShaderProgram::getUniformLocation(const std::string& _uniformName) {
+GLint ShaderProgram::getUniformLocation(const UniformEntries::UniformEntry* _entry) {
```

Comment:
For `Light`s I think we can reliably know the number of light names at scene loading, generate those uniform names and cache then relevant uniforms when the scene just finished loading.